src/firestore/collection/changes.ts

```diff
@@ -28,7 +29,8 @@ export function sortedChanges(query: firebase.firestore.Query, events: firebase.
   return fromCollectionRef(query)
     .map(changes => changes.payload.docChanges)
     .scan((current, changes) => combineChanges(current, changes, events), [])
-    .map(changes => changes.map(c => ({ type: c.type, payload: c })));
+    .map(changes => changes.map(c => ({ type: c.type, payload: c })))
+    .shareReplay(1);
```

Comment:
I think we need `.share()` since `.shareReplay()` caches the data.

Reply:
Without the cache [`it should handle multiple subscriptions (cold)`](https://github.com/angular/angularfire2/pull/1192/files#diff-f36351000f7d9b3fdbd1ea33eaf607deR106) fails.

It's a race condition. I suspect internally in the share the `refCount` has dropped to zero + the share destroyed but it hasn't disposed the `ref` Observable yet. So it results in a miss and a hit, thus never resolving.

In the real world this could be encountered by someone triggering a sequence like so:

1. `collection.take(1).subscribe(...)`
2. something expensive and sync
3. `collection.subscribe(...)`

Less elegant but it would be nice to make it bullet proof.